### ISSUE TITLE
Trying to map shift + arrow keys to select files and move down

### ISSUE THREAD

**Issue by ExoticCandyC** [ranger/ranger#2400]
#### Runtime Environment

- Operating system and version:  Arch
- Terminal emulator and version:  Xterm, latest available on arch repo
- Python version: latest available on arch repo
- Ranger version/commit: latest available on arch repo
- Locale: en-US.UTF8

#### Current Behavior
Trying to map shift+arrow keys (up and down) so i can mark current file and the one below it or above it, just like how it works in graphical file managers. Nothing worked, i spent an hour googling. and non worked.
The ones tested how include:
map <S-DOWN>
map <SHIFT>-<DOWN>
map <SHIFT><DOWN>
(also the lower case versions)

#### Expected Behavior
I expect those key binds to work, but they refuse to work. Maybe it's me being unable to figure it out, and i saw nothing on the manual. I would be happy for guidance if im doing something wrong.

#### Context

I would appreciate to hear how to bind multiple commands to a single key as well, in this case, marking + moving down. didn't see anything anywhere on how to achieve that.

#### Possible Solutions
well, it's a simple key combination. maybe its not defined yet, maybe im doing it wrong and im finding nothing on the manual/forums.

#### Steps to reproduce
1. just set the key bind in the config files and it wont work.

#### Traceback

```

```

**Comment by toonn** [comment]
Your map examples aren't coming through right in the formatting. But them between ``` `backticks` ``` like so.

To map the arrow keys you just use `<up>` and `<down>` as keys, so `map <up> echo Upvote!`.

Sounds like maybe what you want is visual mode though, use v to toggle it.

Chaining multiple commands is done using the `:chain` command.

**Comment by toonn** [comment]
This sort of question is better suited to the [r/ranger subreddit](https://www.reddit.com/r/ranger). There's a lot more people there who'd be happy to help.

**Comment by ExoticCandyC** [comment]
Sorry, forgot to put in the backticks.
Well, im kind of familiar with the keybinds in general, and have already looked through the reddit, and even read the keybinding loader source code to see if im missing something that makes it not work. The problem was everyone, just like pointed here, maps the arrow keys, not them **combined** with a modifier key, which seems to not work.

Things i have tried:
`map <S-DOWN> THE_COMMAND`
`map <SHIFT><DOWN> THE_COMMAND`
`map <SHIFT>-<DOWN> THE_COMMAND`

**Comment by ExoticCandyC** [comment]
Sorry, forgot to put in the backticks.
Well, im kind of familiar with the keybinds in general, and have already looked through the reddit, and even read the keybinding loader source code to see if im missing something that makes it not work. The problem was everyone, just like pointed here, maps the arrow keys, not them **combined** with a modifier key, which seems to not work.

Things i have tried:
`map <S-DOWN> THE_COMMAND`
`map <SHIFT><DOWN> THE_COMMAND`
`map <SHIFT>-<DOWN> THE_COMMAND`

**Comment by toonn** [comment]
Do you use tmux? Because that might be intercepting the keys.

**Comment by ExoticCandyC** [comment]
Nope. The things im using that can intercept the keys (and yet i've double checked to make sure they dont have these key combos bind) are i3wm and the default key bindings for xterm. (im using the xterm itself, but had this problem on gnome-terminal, xfce4-terminal, alacritty too)
If  there is a method to produce a log of input keys, i would be glad to produce and post it.

**Comment by toonn** [comment]
I can reproduce it but I haven't looked into it much. `<s-tab>` does work for example.

**Comment by Twix53791** [comment]
Just to let people know about this small program, nkeypad, [https://github.com/chortlesource/nkeypad](https://github.com/chortlesource/nkeypad), which gives you the keycodes of combined keys like shift-arrows keys. You need [ncurses](https://archlinux.org/packages/core/x86_64/ncurses/.) S-down is 336, S-up 337, s-left 393 and s-right 402 by the way. You can after just use map <402> command in rc.conf.

**Comment by toonn** [comment]
@Twix53791, thanks for that pointer. Maybe what's going on here is that Shift+UP maps to different keycodes in different terminals and that's why it worked with `TERM=linux` but not `TERM=xterm`.
